src/vs/workbench/common/editor/editorInput.ts

```diff
@@ -180,6 +180,14 @@ export abstract class EditorInput extends AbstractEditorInput {
 		return false;
 	}
 
+	/**
+	 * Returns if the input has changes that are not backed up,
+	 * whether or not it is marked dirty.
+	 */
+	hasUnsavedContent(): boolean {
```

Comment:
I have mixed feelings about this one. We could simply introduce a new `EditorInputCapability` called the same as the working copy (`Scratchpad`) and then check on the capability instead?

Reply:
that sounds fine - I was thinking that we wanted information about if there was any content as well, but it looks like we can just get away with checking the capability on its own for now

Reply:
That is a good point but I think its also fine like now.